core/trino-spi/src/main/java/io/trino/spi/connector/ConnectorMetadata.java

```diff
@@ -941,6 +941,8 @@ default Optional<LimitApplicationResult<ConnectorTableHandle>> applyLimit(Connec
      * <b>Note</b>: Implementation must not maintain reference to {@code constraint}'s {@link Constraint#predicate()} after the
      * call returns.
      * </p>
+     *
+     * @param constraint constraint to be applied to the table. {@link Constraint#getSummary()} is guaranteed not to be {@link TupleDomain#isNone() none}.
```

Comment:
Who enforces this? By having PushPredicateIntoTableScan consume the constraint and never reaching `applyFilter`?

Reply:
yes

Reply:
added a check in default impl of `io.trino.spi.connector.ConnectorMetadata#applyFilter` for executable documentation purposes